api_client/python/timesketch_api_client/client.py

```diff
@@ -365,102 +367,232 @@ def _create_session(
+        with a "falsy" JSON payload (e.g., null, empty list/dictionary).
+
 
         Args:
             resource_uri (str): The URI to the resource to be fetched.
-            params (dict): Dict of URL parameters to send in the GET request.
+            params (dict, optional): A dictionary of URL parameters to send
+                in the GET request. Defaults to None.
 
         Returns:
-            Dictionary with the response data.
+            dict: A dictionary containing the JSON response data from the API.
 
         Raises:
-            ValueError: If response could not be JSON-decoded after
-                DEFAULT_RETRY_COUNT attempts.
-            RuntimeError: If the API server returns an error or empty.
+            requests.exceptions.ConnectionError: If a connection error persists
+                after all retry attempts.
+            ValueError: If the API response cannot be JSON-decoded after all
+                retry attempts.
+            RuntimeError: If the API server returns an error (non-20x status code)
+                or a "falsy" JSON response (e.g., null, empty list/dict)
+                after all retry attempts.
         """
         resource_url = "{0:s}/{1:s}".format(self.api_root, resource_uri)
 
-        retry_count = 0
+        # Start attempt count from 0, first loop with set it to 1
+        attempt = 0
         result = None
         while True:
-            retry_count += 1
-            response = self.session.get(resource_url, params=params)
+            attempt += 1
+            # If this is not the first attempt, wait before trying again
+            if attempt > 1:
+                # Simple fixed backoff before retrying
+                time.sleep(1)  # Wait for 1 second before the next attempt
+
             try:
+                response = self.session.get(resource_url, params=params)
                 result = error.get_response_json(response, logger)
                 if result:
                     return result
             except RuntimeError as e:
-                if retry_count >= self.DEFAULT_RETRY_COUNT:
+                if attempt >= self.DEFAULT_RETRY_COUNT:
+                    # Re-raise the original error after exhausting retries
                     raise RuntimeError(
                         "Error for request '{0:s}' - '{1!s}'".format(resource_url, e)
                     ) from e
 
                 logger.warning(
-                    "[{0:d}/{1:d}] Parsing the response for request '{2:s}'"
-                    "failed. Trying again...".format(
-                        retry_count, self.DEFAULT_RETRY_COUNT, resource_url
-                    )
+                    "[%d/%d] API error (RuntimeError) for request '%s' "
+                    "failed. Error: %s. Trying again...",
+                    attempt,
+                    self.DEFAULT_RETRY_COUNT,
+                    resource_url,
+                    e,
```

Comment:
```suggestion
                    str(e),
```

I would recommend to send `e` through `str()` here to ensure it is compatible with `%s`. At least for me in the past, not doing this caused some issues before.